### PR TITLE
Feature pick place exercise ros2 porting

### PR DESCRIPTION
**Pull request by jrgnicho**
This PR contains the port of the pick and place lab exercise, besides porting ros1 code to ros2 the following additional changes have been made:
- Uses MoveItCpp instead of the MoveGroupInterface for all planning tasks inside the pick and place application
- Removal of some old packages (In progress)
- Renaming of packages to be more consistent with the other lab exercises
- Introduces MoveItConfigHelper python class to simplify loading MoveIt2 configuration files and MoveIt2 nodes, this may eventually be moved to a separate location/repo
Still to do are 
- Testing on actual robot
- Updating exercise comments
- Update the template version
- Update wiki, perhaps to be addressed in a separate PR

### CODE REVIEW

**Comment by jrgnicho** [comment]
Added more changes, a quick summary of those are as follows:
- Pick and place application is fully functional in virtual mode only.
- Removed deprecated packages and several ros1 files (mostly launch)
- Added source dependency lists which can be loaded with vcs-tools

**Comment by jrgnicho** [comment]
@jdlangs this branch has been tested on real hardware, UR3 CB3 series so it's ready for review.  Updating the comments and related documentation will be done in a separate PR

**Comment by jrgnicho** [comment]
This PR works in galactic and with moveit from debian, it may not work in foxy.

**Comment by jdlangs** [comment]
I can confirm this works in simulation on Foxy with just a few trivial changes. I don't see any reason why the physical hardware wouldn't work as well.

We should definitely target foxy as it greatly simplifies the dependency situation in addition to it being supported longer than galactic.

**Comment by jdlangs** [comment]
I noticed this wipes out the ROS1 demo, I think we should continue to have both available for now. Probably we can do the same thing as the exercises and push them down into separate `ros1/` and `ros2/` subfolders.

**Comment by jdlangs** [comment]
And while we're at it, I say we rename the folder to something more obvious (e.g., `demo1`)

File: exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos [hunk 1]
```diff
@@ -0,0 +1,26 @@
+repositories:
```
Comment (jdlangs) [inline exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos]: I forgot I had to build [pcl_ros](https://github.com/ros-perception/perception_pcl/tree/ros2) as well to make this work on foxy.

File: exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    version: foxy-devel
+  ros2_control:
+    type: git
+    url: https://github.com/ros-controls/ros2_control.git
+    version: foxy
+  ros2_controllers:
+    type: git
+    url: https://github.com/ros-controls/ros2_controllers.git
+    version: foxy
+
```
Comment (jdlangs) [inline exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos]: Can remove the controller stuff as source dependencies.

File: exercises/Perception-Driven_Manipulation/solution_ws/src/moveit2_dependencies.repos [hunk 3]
```diff
@@ -0,0 +1,25 @@
+repositories:
```
Comment (jdlangs) [inline exercises/Perception-Driven_Manipulation/solution_ws/src/moveit2_dependencies.repos]: We can just delete these files, yes?

Comment (jrgnicho) [reply exercises/Perception-Driven_Manipulation/solution_ws/src/moveit2_dependencies.repos]: For foxy yes

File: exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos (hunk 2 above)
Comment (jrgnicho) [reply exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos]: Yes

File: exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos (hunk 1 above)
Comment (jrgnicho) [reply exercises/Perception-Driven_Manipulation/solution_ws/src/dependencies.repos]: OK, I'll add it to the dependency list

**Comment by jrgnicho** [comment]
> I noticed this wipes out the ROS1 demo, I think we should continue to have both available for now. Probably we can do the same thing as the exercises and push them down into separate `ros1/` and `ros2/` subfolders.

Agreed, can you think of an easy way to restore the ROS1 version?

**Comment by jdlangs** [comment]
> Agreed, can you think of an easy way to restore the ROS1 version?

I would move the changed folder into a `ros2` subfolder and then copy the original demo back in a from an unmodified version. Not sure if there's a more direct git method.

**Comment by jrgnicho** [comment]
> > Agreed, can you think of an easy way to restore the ROS1 version?
> 
> I would move the changed folder into a `ros2` subfolder and then copy the original demo back in a from an unmodified version. Not sure if there's a more direct git method.

OK I'll do that, I can't think of another way either

**Comment by jrgnicho** [comment]
@jdlangs I made the requested changes.  I think we should merge #350 so that we can then copy the ros1 pick and place code under the ros1 directory

**Comment by jdlangs** [comment]
After merging #350 there was some intense git surgery needed so this would appear as a new `ros2` subfolder. Since I rebased this I'm opening a new pull request rather than overwriting this branch. The original commit history will be preserved though.

**Pull request closed without merging**